### PR TITLE
UCP/CONTEXT: fix CM initialization

### PR DESCRIPTION
**Pull request by evgeny-leksikov**
## What
fixes https://github.com/openucx/ucx/issues/4808

Update: all commits related to close protocol are moved to https://github.com/openucx/ucx/pull/5075 to avoid merge conflicts

### CODE REVIEW

**Comment by evgeny-leksikov** [comment]
bot:pipe:retest

**Comment by evgeny-leksikov** [comment]
bot:pipe:retest

File: src/ucp/core/ucp_context.c [hunk 1]
```diff
@@ -1031,6 +1031,13 @@ static void ucp_fill_sockaddr_prio_list(ucp_context_h context,
                                     num_sockaddr_tls);
     ucp_fill_sockaddr_cms_prio_list(context, sockaddr_tl_names,
                                     num_sockaddr_tls, sockaddr_cm_enable);
+    if ((context->config.ext.sockaddr_cm_enable == UCS_YES) &&
+        (context->config.num_cm_cmpts == 0)) {
+        ucs_error("UCX_SOCKADDR_CM_ENABLE is set to yes but no one component supports CM");
```
Comment (alinask) [inline src/ucp/core/ucp_context.c:1036]: "UCX_SOCKADDR_CM_ENABLE is set to yes but none of the available components supports CM"
can we print the available components to the user to make this message more informative?

Comment (evgeny-leksikov) [reply src/ucp/core/ucp_context.c:1036]: do you think list of components is useful even if none of them supports required capability?

Comment (alinask) [reply src/ucp/core/ucp_context.c:1036]: you say in the message that none of the components supports CM. do you think the user knows which components we tested for it?

Comment (evgeny-leksikov) [reply src/ucp/core/ucp_context.c:1036]: He is trying to enable UCX_SOCKADDR_CM_ENABLE explicitly, IMO message like his system does not support this feature is enough.

Comment (alinask) [reply src/ucp/core/ucp_context.c:1036]: this value (yes) will become the default soon so it won't be set explicitly. 
IMO it's better to provide more useful information when it's possible, not only for the user but also for us when we will be debugging.

Comment (evgeny-leksikov) [reply src/ucp/core/ucp_context.c:1036]: > this value (yes) will become the default soon

Should not "try" to be set by default?
> to provide more useful information when it's possible

list of present components is useless, do you think we have to point to required SW & HW? ~~link to the market?~~ :)

File: src/ucp/wireup/wireup_cm.c [hunk 2]
```diff
@@ -941,10 +947,19 @@ ucp_request_t* ucp_ep_cm_close_request_get(ucp_ep_h ep)
...
+    if (elem->cb == ucp_cm_client_connect_progress) {
+        client_connect_arg = elem->arg;
+        if (client_connect_arg->ucp_ep == arg) {
+            ucp_cm_client_connect_prog_arg_free(client_connect_arg);
+            return 1;
+        } else {
+            return 0;
+        }
+    } else if ((elem->cb == ucp_ep_cm_disconnect_progress)        ||
+               (elem->cb == ucp_ep_cm_remote_disconnect_progress) ||
```
[the remark this answers is not in the document]
Comment (evgeny-leksikov) [reply src/ucp/wireup/wireup_cm.c:961]: will fix here https://github.com/openucx/ucx/pull/5075
I'm going to remove cherry-picked commits from this PR

[the remark this answers is not in the document]
Comment (evgeny-leksikov) [reply src/ucp/wireup/wireup_cm.c:961]: #5075 / 0b2a28b10f3a4e550912769d907177acbed0dfc5 fixes this

**Pull request merged**